IdentityCore/src/workplacejoin/ios/MSIDWorkPlaceJoinUtil.m

```diff
@@ -29,140 +29,108 @@
-    CHECK_KEYCHAIN_STATUS(@"retrieve wpj identity attr");
     
-    cerDict = (__bridge NSDictionary *) result;
-    assert([cerDict isKindOfClass:[NSDictionary class]]);
-    issuer = [cerDict objectForKey:(__bridge id)kSecAttrIssuer];
-    certificateIssuer = [[NSString alloc] initWithData:issuer encoding:NSISOLatin1StringEncoding];
-    CFRelease(result);
-    result = NULL;
+    MSID_LOG_VERBOSE(nil, @"Attempting to get registration information - shared access Group");
+    MSID_LOG_VERBOSE_PII(nil, @"Attempting to get registration information - %@ shared access Group", sharedAccessGroup);
     
-    // now get the identity out and use it.
-    [identityAttr removeObjectForKey:(__bridge id<NSCopying>)(kSecReturnAttributes)];
-    status = SecItemCopyMatching((__bridge CFDictionaryRef)identityAttr, (CFTypeRef*)&identity);
-    CHECK_KEYCHAIN_STATUS(@"retrieve wpj identity ref");;
-    if (CFGetTypeID(identity) != SecIdentityGetTypeID())
+    identity = [self copyWPJIdentity:context sharedAccessGroup:sharedAccessGroup certificateIssuer:&certificateIssuer];
+    if (!identity || CFGetTypeID(identity) != SecIdentityGetTypeID())
     {
-        CFRelease(identity);
-        
-        if (error)
-        {
-            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Wrong object type returned from identity query", nil, nil, nil, context.correlationId, nil);
-        }
+        MSID_LOG_VERBOSE(context, @"Failed to retrieve WPJ identity.");
+        CFReleaseNull(identity);
         return nil;
     }
-    //Get the certificate and data
+    
+    // Get the wpj certificate
+    MSID_LOG_VERBOSE(context, @"Retrieving WPJ certificate reference.");
     status = SecIdentityCopyCertificate(identity, &certificate);
-    CHECK_KEYCHAIN_STATUS(@"copy identity certificate");
     
+    // Get the private key
+    MSID_LOG_VERBOSE(context, @"Retrieving WPJ private key reference.");
     status = SecIdentityCopyPrivateKey(identity, &privateKey);
-    CHECK_KEYCHAIN_STATUS(@"copy identity private key");
     
     certificateSubject = (NSString *)CFBridgingRelease(SecCertificateCopySubjectSummary(certificate));
     certificateData = (NSData *)CFBridgingRelease(SecCertificateCopyData(certificate));
     
-    if(!(identity && certificate && certificateSubject && certificateData && privateKey && certificateIssuer))
+    if(!(certificate && certificateSubject && certificateData && privateKey && certificateIssuer))
     {
         // We never should hit this error anyways, as any of this stuff being missing will cause failures farther up.
         if (error)
         {
-            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Missing some piece of WPJ data", nil, nil, nil, context.correlationId, nil);
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Missing some pieces of WPJ data", nil, nil, nil, context.correlationId, nil);
```

Comment:
should we release some of the objects if we hit error as well?

Reply:
Done.